modules/cudaoptflow/src/pyrlk.cpp

```diff
@@ -174,9 +170,14 @@ namespace
         if (err)
             ensureSizeIsEnough(1, prevPts.cols, CV_32FC1, *err);
 
-        if (prevPyr.size() != size_t(maxLevel_ + 1) || nextPyr.size() != size_t(maxLevel_ + 1))
+        if (prevPyr.size() != size_t(maxLevel_ + 1))
         {
-            buildImagePyramid(prevPyr[0], prevPyr, nextPyr[0], nextPyr, stream);
+            buildImagePyramid(prevPyr[0], prevPyr, maxlevel_, stream);
+        }
+
+        if (nextPyr.size() != size_t(maxLevel_ + 1))
+        {
+            buildImagePyramid(nextPyr[0], nextPyr, maxlevel_, stream);
```

Comment:
Actually this may be slower (more synchronization points) - should be tested

Reply:
I will take a look into it.

Reply:
I thought about this, went through the code and observed that all the operations are issued to the same stream in CUDA which was happening earlier too. I will test it too to verify.
So, I think that there will not be any issues with the performance. However, please let me know if I have misinterpreted something.
Thank you.

Reply:
@alalek What do you think of this?